src/AnimatedFade.tsx

```diff
@@ -1,52 +1,108 @@
-import React, { useEffect, useRef } from 'react';
+import React, { useCallback, useEffect, useRef } from 'react';
 import { Animated, ViewStyle } from 'react-native';
+import useCurrent from './hooks/useCurrent';
 
 type Props = {
-  fadeOnMount?: boolean;
+  /**
+   * auto-trigger fade-in?
+   */
   triggerFade?: boolean;
-  startingOpacity?: number; // 0 - 1.00
-  endingOpacity?: number; // 0 - 1.00
-  duration?: number; // ms
-  delay?: number; // ms
+
+  /**
+   * starting opacity value (0 - 1.00)
+   */
+  opacityStart?: number;
+
+  /**
+   * ending opacity value (0 - 1.00)
+   */
+  opacityEnd?: number;
+
+  /**
+   * animation duration (in ms)
+   */
+  duration?: number;
+
+  /**
+   * delay before animation start (in ms)
+   */
+  delay?: number;
   style?: ViewStyle;
+
+  /**
+   * callback function called once fadeIn animation completes
+   */
   onEnd?: () => void;
+
+  /**
+   * should this animation repeat?
+   */
+  loop?: boolean;
 };
 
 const AnimatedFade: React.FC<Props> = ({
-  fadeOnMount = true,
-  triggerFade = false,
-  startingOpacity = 0,
-  endingOpacity = 1,
+  triggerFade = true,
+  opacityStart = 0,
+  opacityEnd = 1,
   duration = 800,
   delay = 0,
+  loop = false,
   onEnd,
   style,
   children,
 }) => {
-  const opacity = useRef(new Animated.Value(startingOpacity)).current;
+  const opacity = useRef(new Animated.Value(opacityStart));
```

Comment:
Capturing the ref allows for better control if the dev needs to pass `opacity` into a useEffect dependency array (otherwise, passing `opacity` into a dependency array would result in an infinite loop)